tensorflow/core/graph/mkl_layout_pass.cc

```diff
@@ -1915,9 +1915,9 @@ void MklLayoutRewritePass::GetDummyMklTensorNode(std::unique_ptr<Graph>* g,
     Node* orig_input0 = nullptr;
     TF_CHECK_OK(
         orig_node->input_node(0, const_cast<const Node**>(&orig_input0)));
-    // Allow duplicate while adding control edge as it would fail (return
-    // NULL) if we try to add duplicate edge.
-    CHECK_NOTNULL((*g)->AddControlEdge(orig_input0, *out, true));
+    if (!DoesControlEdgeExist(orig_input0, *out)) {
+      (*g)->AddControlEdge(orig_input0, *out, false);
+    }
```

Comment:
If you don't need to check for `nullptr` anymore (in case a unique edge failed to be added), you can just remove `DoesControlEdgeExist`.

Otherwise, adding first and only check when `AddControlEdge` returns `nullptr` should be cheaper since duplicate control edges shouldn't happen.
```
auto edge = (*g)->AddControlEdge(e->src(), new_node, false);
DCHECK(g != nullptr || DoesControlEdgeExist(e->src(), new_node);
```

Reply:
changed. thanks!